### ISSUE TITLE
Cityscape again

### ISSUE THREAD

**Issue by ehabhelmy82** [mrharicot/monodepth#149]
Dear Author
Thanks for you work and your answers
Concerning training on cityscape and testing on KITTI results that mentioned in table 1.
I trained the model three times and every time I get a very far result from that mentioned in the paper.I do the following:
First I extract Cityscape files then put all training images (left and right) in one folder called CS_Train but the image 'troisdorf_000000_000073_rightImg8bit' was black and white so I replace it with 'troisdorf_000000_000073_leftImg8bit'
Then use the following excution commands:
python3 monodepth_main.py --mode train --model_name CSmodel --data_path /data/ehab/CS_Train/ --filenames_file ./utils/filenames/cityscapes_train_files.txt --log_directory ~/CSd/

python3 monodepth_main.py --mode test --data_path /data/ehab/data_scene_flow/ --filenames_file ./utils/filenames/kitti_stereo_2015_test_files.txt --log_directory ~/CSd/ --checkpoint_path ~/CSd/CSmodel/model-143600

python utils/evaluate_kitti.py --split kitti --predicted_disp_path ../CSd/CSmodel/disparities.npy --gt_path ../data/data_scene_flow

I got the following results :
abs_rel,     sq_rel,        rms,    log_rms,     d1_all,         a1,         a2,         a3
    0.2726,     3.9903,      9.041,      0.317,     62.491,      0.661,      0.877,      0.950

When repeated again all three commands I got the following results
abs_rel,     sq_rel,        rms,    log_rms,     d1_all,         a1,         a2,         a3
    0.2750,     3.9985,      9.033,      0.315,     62.641,      0.654,      0.876,      0.951

In the third run I got the following :
abs_rel,     sq_rel,        rms,    log_rms,     d1_all,         a1,         a2,         a3
    0.2587,     3.5841,      8.759,      0.306,     59.505,      0.673,      0.882,      0.953

I can not find out what is wrong? can you help me?
By the way, when I run using the pretrained model I got results very close to that in the paper:
abs_rel,     sq_rel,        rms,    log_rms,     d1_all,         a1,         a2,         a3
    0.7058,    10.4943,     14.665,      0.546,     94.768,      0.054,      0.325,      0.856
I really appreciate your reply

**Comment by ehabhelmy82** [comment]
Thank you very much for your reply
Am not focusing on certain result rather than I want to repeat all reported results exactly so that I can compare with my proposed approach, I almost finished all except for Cityscape and Stereo experiment still not the same results as yours. After your comment I wish it will work well for Cityscape and in parallel am trying to fix Stereo model.
Once again thank you very much

**Comment by ehabhelmy82** [comment]
I used the following:
python3 monodepth_main.py --mode train --model_name CSmodel3 --data_path /data/ehab/CS_Train/ --filenames_file ./utils/filenames/cityscapes_train_files.txt --log_directory ~/CSd3/ --dataset cityscapes

python3 monodepth_main.py --mode test --data_path /data/ehab/data_scene_flow/ --filenames_file ./utils/filenames/kitti_stereo_2015_test_files.txt --log_directory ~/CSd3/ --checkpoint_path ~/CSd3/CSmodel3/model-143600 --dataset cityscapes

python utils/evaluate_kitti.py --split kitti --predicted_disp_path ../CSd3/CSmodel3/disparities.npy --gt_path ../data/data_scene_flow

I got the following:
/home/ehab/monodepth-master/utils/evaluation_utils.py:62: RuntimeWarning: divide by zero encountered in divide
  pred_depth = width_to_focal[width] * 0.54 / pred_disp
/home/ehab/monodepth-master/utils/evaluation_utils.py:62: RuntimeWarning: overflow encountered in divide
  pred_depth = width_to_focal[width] * 0.54 / pred_disp
   abs_rel,     sq_rel,        rms,    log_rms,     d1_all,         a1,         a2,         a3
    6.0278,   418.4342,     64.898,      1.888,    100.000,      0.009,      0.027,      0.057
What is wrong??

**Comment by mrharicot** [comment]
Hi,
You only need the dataset flag when you train, not when you test because you are testing on kitti.

**Comment by ehabhelmy82** [comment]
python3 monodepth_main.py --mode test --data_path /data/ehab/data_scene_flow/ --filenames_file ./utils/filenames/kitti_stereo_2015_test_files.txt --log_directory ~/CSd3/ --checkpoint_path ~/CSd3/CSmodel3/model-143600

python utils/evaluate_kitti.py --split kitti --predicted_disp_path ../CSd3/CSmodel3/disparities.npy --gt_path ../data/data_scene_flow

/home/ehab/monodepth-master/utils/evaluation_utils.py:62: RuntimeWarning: divide by zero encountered in divide
  pred_depth = width_to_focal[width] * 0.54 / pred_disp
/home/ehab/monodepth-master/utils/evaluation_utils.py:62: RuntimeWarning: overflow encountered in divide
  pred_depth = width_to_focal[width] * 0.54 / pred_disp
   abs_rel,     sq_rel,        rms,    log_rms,     d1_all,         a1,         a2,         a3
    6.0278,   418.4342,     64.898,      1.888,    100.000,      0.009,      0.027,      0.057
Still wrong???
Any idea?
Thanks

**Comment by ehabhelmy82** [comment]
yes

**Comment by a-jahani** [comment]
I remember I was getting something similar to you. Whatever I did the result was far away and didn't change at all. The warning divided by zero is important! I think what I was doing was providing it with wrong files that didn't exist!( a typo in my paths) 
make sure you are giving it a correct path!

**Comment by ehabhelmy82** [comment]
These are three out puts one using pretrained model and two using built models from scratch
file:///C:/Users/Ehab2018/Desktop/troisdorf_disp%20CSD1.png
file:///C:/Users/Ehab2018/Desktop/troisdorf.png
file:///C:/Users/Ehab2018/Desktop/troisdorf_disp%20CSD.png
file:///C:/Users/Ehab2018/Desktop/troisdorf_disp%20Pretrained.png

**Comment by abhiagwl4262** [comment]
How do I evaluate performance on cityscape test data? Cityscape disparity ranges from 0-126 while monodepth output range is different.

**Comment by mrharicot** [comment]
Cityscapes doesn't have good quality depth data, they ran an off the shelf stereo algorithm.
As far as I know, no single frame depth estimation paper ran numbers on this dataset.

**Comment by abhiagwl4262** [comment]
I tried to train on cityscape data changing the resolution 512*1024 without cropping car hood. I used pretrained city2kitti_resnet model that is trained on 256*512. results are absolutely rubbish. Training and validation loss  were both decreasing. any insights?

Actually I want to find disparity for the images with resolution 480*640. any suggestions?

**Comment by mrharicot** [comment]
Hi,
Can you give more details as to how the results were bad?
How did the depthmaps look like?
Additionnaly, it doesn't make much sense to reuse the city2kitti model, I would use the cityscapes model instead.

**Comment by abhiagwl4262** [comment]
I will share few disparity maps with you. I am running few more experiments
and will talk to you more about this after training is done.

**Comment by mrharicot** [comment]
Hi,
The network seems to be unable to learn.
1024x512 is really large, why are you trying such high resolution?
I would first try to lower the learning rate to see if you can achieve stable training to begin with.
If you want to use the pretrained model, I would try to use progressively higher resolutions, not directly  1024x512.
